pkg/apis/openebs.io/scp/v1alpha1/storage-volume.go

```diff
@@ -39,3 +44,266 @@ type StorageVolumeList struct {
+	StorageTier StorageTier `json:"storageTier"`
+
+	DataProtection *DataProtectionCapabilities `json:"dataProtection,omitempty"`
+}
+
+type StorageTier string
+
+const (
+	// StorageTierPlatinum indicates that the tier to be used for volume is platinum
+	StorageTierPlatinum StorageTier = "P1"
+
+	// StorageTierGold indicates that the tier to be used for volume is gold
+	StorageTierGold StorageTier = "P2"
+
+	// StorageTierSilver indicates that the tier to be used for volume is silver
+	StorageTierSilver StorageTier = "P3"
+)
+
+// DataStorageCapabilities defines data storage capabilities for a volume
+type DataStorageCapabilities struct {
+	//AccessModes contains the actual access modes for the volume
+	AccessModes StorageVolumeAccessMode `json:"accessModes"`
+
+	// ProvisioningPolicy defines provisioning policy type
+	ProvisioningPolicy ProvisioningPolicy `json:"provisioningPolicy"`
+
+	// RecoveryTimeObjectives to be supported or not
+	RecoveryTimeObjectives RecoveryTimeObjectives `json:"recoveryTimeObjectives"`
+
+	// Compression to be supported or not, if yes the algorithm
+	//	+nullable
+	Compression *CompressionAlgorithm `json:"compression,omitempty"`
+
+	// Deduplication to be supported or not
+	// +nullable
+	Deduplication *bool `json:"deduplication,omitempty"`
+}
+
+type StorageVolumeAccessMode string
+
+const (
+	// can be mounted in read/write mode to exactly 1 host
+	ReadWriteOnce StorageVolumeAccessMode = "ReadWriteOnce"
+	// can be mounted in read-only mode to many hosts
+	ReadOnlyMany StorageVolumeAccessMode = "ReadOnlyMany"
+	// can be mounted in read/write mode to many hosts
+	ReadWriteMany StorageVolumeAccessMode = "ReadWriteMany"
+)
+
+
+type RecoveryTimeObjectives string
+
+const (
+	// OnlineActive indicates that the recovery time objective for a volume is online active
+	OnlineActive RecoveryTimeObjectives = "OnlineActive"
+
+	// OnlinePassive indicates that the recovery time objective for a volume is online active
+	OnlinePassive RecoveryTimeObjectives = "OnlinePassive"
+)
+
+type ProvisioningPolicy string
+
+const (
+	// ProvisioningPolicyThick indicates that the provisioning policy is thick
+	ProvisioningPolicyThick ProvisioningPolicy = "Thick"
+
+	// ProvisioningPolicyThin indicates that the provisioning policy is thin
+	ProvisioningPolicyThin ProvisioningPolicy = "Thin"
+)
+
+type CompressionAlgorithm string
+
+const(
+	HuffmanCompression CompressionAlgorithm = "huffman"
+)
+
+type DataSecurityCapabilities struct {
+	MediaEncryption *bool `json:"mediaEncryption,omitempty"`
+}
+
+type IOConnectivityCapabilities struct {
+	AccessProtocols string `json:"accessProtocols,omitempty"`
+}
+
+type IOPerformanceCapabilities struct {
+	AverageIOOperationLatencyMicroseconds string `json:"averageIOOperationLatencyMicroseconds,omitempty"`
+}
+
+type DataProtectionCapabilities struct {
+	// TODO: Decide the Fields
+}
+
+// Affinity is a group of affinity scheduling rules
+type Affinity struct {
```

Comment:
Can affinity be move to a different file just like capabilities? cc @shovanmaity

Reply:
It is used only in volumes. Though we can do it